v2/samples/direct_method_invocation_sample/README.md

```diff
@@ -0,0 +1,81 @@
+Azure IoT Edge V2 - Direct Method Invocation Sample (Linux only for now) 
+=====================================================================
+
+Overview
+--------
+
+This sample showcases how one can invoke a direct method from one module to another module 
+or a downstream device on [Azure IoT Edge V2](https://docs.microsoft.com/en-us/azure/iot-edge/how-iot-edge-works). 
+
+It uses the following simple scenario -
+There is an edge deployment consisting of 2 custom modules -
+1. An engine simulator module produces data which includes the engine RPM. It also exposes 
+a Reset method, which if called, resets the engine to its initial value.
+2. An analyzer module analyzes the data sent by the engine simulator module and invokes a 
+Reset direct method call on the engine simulator module if the RPM value is too high.
+
+Invoking direct methods
+--------------
+
+A module can invoke a direct method on another module or a downstream device (i.e. a device
+connected to the same EdgeHub), by using its EdgeHubConnectionString. It cannot invoke direct methods
+on modules on other edge devices, or devices not connected to the same EdgeHub (unless it uses an 
+IotHub scoped connection string).
+
+Here are the steps to invoke a direct method from one module to another -
+
+- You need to use the ServiceClient to invoke direct methods from a module. The ServiceClient is in the Microsoft.Azure.Devices nuget package which is not added by default, so you need to manually add that. Make sure to add a reference to a preview* version of the nuget package, to get modules support. 
+
+- Create a ServiceClient using the EdgeHubConnectionString that is passed in to the module as an 
+environment variable. 
+```
+ServiceClient serviceClient = ServiceClient.CreateFromConnectionString(connectionString);
+```
+> Note: **Do not** call OpenAsync on the ServiceClient when initializating it with the EdgeHubConnectionString.
+This will cause it to try to open the AMQP connection which is not supported, causing it to throw an exception. Method invocation happens over HTTP. 
+
+- Invoke direct method on another module like this -
+```
+var response = await serviceClient.InvokeDeviceMethodAsync(message.ConnectionDeviceId, message.ConnectionModuleId, new CloudToDeviceMethod("Reset"));                    
+```
+> Note: An incoming message contains the device Id in the property ConnectionDeviceId, and the
+module Id in the property ConnectionModuleId (if the message is from another module). You can 
+use those values to invoke a method on that device or module.
+
+- Receiving direct method invocations from a module is same as if receiving it from the cloud. 
```

Comment:
For completeness, please include what that looks like here, for folks who haven't seen it yet.

Reply:
Sure, makes sense.